sgkit/stats/popgen.py

```diff
@@ -682,3 +688,190 @@ def pbs(
         {variables.stat_pbs: (["windows", "cohorts_0", "cohorts_1", "cohorts_2"], p)}
     )
     return conditional_merge_datasets(ds, variables.validate(new_ds), merge)
+
+
+N_GARUD_H_STATS = 4  # H1, H12, H123, H2/H1
+
+
+def _Garud_h(k: ArrayLike) -> ArrayLike:
```

Comment:
I'm having trouble understanding what ``k`` is here - any chance of a more descriptive name?

Unless k is quite small, isn't ``sorted(collections.Counter(k.tolist()).values(), reverse=True)`` going to be bottleneck?

Reply:
Changed to `haplotypes`.

I thought that the sorted call would be a bottleneck too, but I haven't seen that during my MalariaGEN testing.